iml-agent/src/daemon_plugins/postoffice.rs

```diff
@@ -173,11 +175,14 @@ impl DaemonPlugin for PostOffice {
     }
 
     async fn teardown(&mut self) -> Result<(), ImlAgentError> {
-        if let Some(wd) = self.wd.lock().await.0.clone() {
-            let _ = self.inotify.lock().await.rm_watch(wd);
-        }
         // drop all triggers
         self.routes.lock().await.clear();
+        let inotify = std::mem::replace(&mut *self.inotify.lock().await, None);
+        inotify
+            .unwrap()
+            .close()
```

Comment:
> When all file descriptors referring to an inotify instance have been closed, the underlying object and its resources are freed for reuse by the kernel; all associated watches are automatically freed. --- https://linux.die.net/man/7/inotify

Reply:
Does this solve the issue you are seeing?

Reply:
Well, yes. But not all the issues like never-ending loops.